ext/mbstring/libmbfl/filters/mbfilter_sjis_mobile.c

```diff
@@ -176,467 +178,435 @@ const unsigned short mbfl_kddi2uni_pua_b[8][3] = {
 		}
 	}
-
-	return w;
+	return 0;
 }
 
-int
-mbfilter_sjis_emoji_kddi2unicode(int s, int *snd)
-{
-	int w = s, si, c;
-	const int nflags_order_kddi[] = {3, 1, 5, 4, 0, 7};
+#define EMIT_FLAG_EMOJI(country) \
+	*snd = NFLAGS((country)[0]); \
+	return NFLAGS((country)[1])
+
+static const char nflags_kddi[6][2] = {"FR", "DE", "IT", "GB", "CN", "KR"};
 
-	*snd = 0;
+int mbfilter_sjis_emoji_kddi2unicode(int s, int *snd)
+{
 	if (s >= mb_tbl_code2uni_kddi1_min && s <= mb_tbl_code2uni_kddi1_max) {
-		si = s - mb_tbl_code2uni_kddi1_min;
-		if (si == 0x0008) { /* ES */
-			*snd = NFLAGS(nflags_s[2][0]); w = NFLAGS(nflags_s[2][1]);
-		} else if (si == 0x0009) { /* RU */
-			*snd = NFLAGS(nflags_s[8][0]); w = NFLAGS(nflags_s[8][1]);
-		} else if (si >= 0x008d && si <= 0x0092) {
-			c = nflags_order_kddi[si-0x008d];
-			*snd = NFLAGS(nflags_s[c][0]); w = NFLAGS(nflags_s[c][1]);
-		} else if (si == 0x0104) {
-			*snd = 0x0023; w = 0x20E3;
+		if (s == 0x24C0) { /* Spain */
+			EMIT_FLAG_EMOJI("ES");
+		} else if (s == 0x24C1) { /* Russia */
+			EMIT_FLAG_EMOJI("RU");
+		} else if (s >= 0x2545 && s <= 0x254A) {
+			EMIT_FLAG_EMOJI(nflags_kddi[s - 0x2545]);
+		} else if (s == 0x25BC) {
+			EMIT_KEYPAD_EMOJI('#');
 		} else {
-			w = mb_tbl_code2uni_kddi1[si];
-			if (w > 0xf000) {
-				w += 0x10000;
-			} else if (w > 0xe000) { /* unsupported by Unicode 6.0 */
-				w += 0xf0000;
-			}
+			*snd = 0;
+			return convert_emoji_cp(mb_tbl_code2uni_kddi1[s - mb_tbl_code2uni_kddi1_min]);
 		}
 	} else if (s >= mb_tbl_code2uni_kddi2_min && s <= mb_tbl_code2uni_kddi2_max) {
-		si = s - mb_tbl_code2uni_kddi2_min;
-		if (si == 100) { /* JP */
-			*snd = NFLAGS(nflags_s[6][0]); w = NFLAGS(nflags_s[6][1]);
-		} else if (si >= 0x00ba && si <= 0x00c2) {
-			*snd = si-0x00ba+0x0031; w = 0x20E3;
-		} else if (si == 0x010b) { /* US */
-			*snd = NFLAGS(nflags_s[9][0]); w = NFLAGS(nflags_s[9][1]);
-		} else if (si == 0x0144) {
-			*snd = 0x0030; w = 0x20E3;
+		if (s == 0x2750) { /* Japan */
+			EMIT_FLAG_EMOJI("JP");
+		} else if (s >= 0x27A6 && s <= 0x27AE) {
+			EMIT_KEYPAD_EMOJI(s - 0x27A6 + '1');
+		} else if (s == 0x27F7) { /* United States */
+			EMIT_FLAG_EMOJI("US");
+		} else if (s == 0x2830) {
+			EMIT_KEYPAD_EMOJI('0');
 		} else {
-			w = mb_tbl_code2uni_kddi2[si];
-			if (w > 0xf000) {
-				w += 0x10000;
-			} else if (w > 0xe000) { /* unsupported by Unicode 6.0 */
-				w += 0xf0000;
-			}
+			*snd = 0;
+			return convert_emoji_cp(mb_tbl_code2uni_kddi2[s - mb_tbl_code2uni_kddi2_min]);
 		}
 	}
-	return w;
+	return 0;
 }
 
-int
-mbfilter_sjis_emoji_sb2unicode(int s, int *snd)
-{
-	int w = s, si, c;
-	const int nflags_order_sb[10] = {6, 9, 3, 1, 5, 4, 2, 8, 0, 7};
+static const char nflags_sb[10][2] = {"JP", "US", "FR", "DE", "IT", "GB", "ES", "RU", "CN", "KR"};
 
-	*snd = 0;
+int mbfilter_sjis_emoji_sb2unicode(int s, int *snd)
+{
 	if (s >= mb_tbl_code2uni_sb1_min &&	s <= mb_tbl_code2uni_sb1_max) {
```

Comment:
There's a weird tab here after the `&&`.

Reply:
It sure is weird! I wonder what happened?